definitions/npm/cookie-parser_v1.x.x/flow_v0.25.x-/cookie-parser_v1.x.x.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * Flow libdef for 'cookie-parser'
+ * See https://www.npmjs.com/package/cookie-parser
+ * by Vincent Driessen, 2018-12-21
+ */
+
+declare module 'cookie-parser' {
+  declare function cookieParser(
+    secret?: string | Array<string>,
+    options?: mixed,
+  ): express$Middleware;
```

Comment:
Mostly just to learn about best practices here… how would one go about typing this return type?  It should be equal to whatever this is:
https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/express_v4.x.x/flow_v0.32.x-/express_v4.x.x.js#L103-L105

How can one refer to a type defined elsewhere? What's the community best practice here?

Reply:
Right now the best practice is unfortunately to copy the relevant types over. The CLI cannot resolve type dependencies. See e.g this issue for further details: https://github.com/flow-typed/flow-typed/issues/1857

Reply:
OK, thanks for providing some background material here.  For now, I've copied over the necessary type defs from `express` manually just to try how that would work (see ba756cc1975bcf39fd38a07c5145ae65b8e86cff), but it's indeed hard to see how this will scale.  It seems very laborious to keep updating all libdefs that depend on express when the express annotations get updated…